File: src/main/java/com/back/domain/member/member/enums/Role.java
package com.back.domain.member.member.enums;

public enum Role {
    ADMIN,
    USER;

    public static Role from(String value) {
        if (value == null) {
            throw new IllegalArgumentException("Role is null");
        }

        try {
            return Role.valueOf(value.toUpperCase());
        } catch (IllegalArgumentException e) {
            throw new IllegalArgumentException("Invalid role: " + value);
        }
    }
}

File: src/main/java/com/back/domain/member/member/enums/Role.kt
package com.back.domain.member.member.enums

enum class Role {
    ADMIN,
    USER;

    companion object {
        @JvmStatic
        fun from(value: String?): Role {
            requireNotNull(value) { "Role is null" }

            return runCatching { valueOf(value.uppercase()) }  // [1] Locale 파라미터 불필요
                .getOrElse { throw IllegalArgumentException("Invalid role: $value") }
        }
    }
}
